crates/wasi-common/src/sys/unix/bsd/hostcalls_impl.rs

```diff
@@ -167,3 +174,12 @@ pub(crate) mod fd_readdir_impl {
         Ok(dir.lock().unwrap())
     }
 }
+
+/// Trim trailing slash characters from the given `HostString`.
+fn trim_end_slashes(host: HostString) -> HostString {
+    let mut bytes = host.into_bytes();
+    while let Some(b'/') = bytes.last() {
+        bytes.pop();
+    }
+    unsafe { CString::from_vec_unchecked(bytes) }
+}
```

Comment:
Given how many different helper methods for `HostString` we've got scattered here and there, I'm really leaning towards having it as a transparent struct `HostString(CString)` and have those methods be the implementation on the new transparent type.